### PR TITLE
Add snap install instructions

### PR DESCRIPTION
**Pull request by flexiondotorg**
This pull request adds installation instructions for users of [snap enabled Linux distributions](https://snapcraft.io/docs/core/install). Snaps enable ISVs and projects to make their software available to millions of Linux users via the snap store and directly control delivery of software updates to their users. Snaps of stable desktop applications will also be available via the desktop Software Center, improving the discoverability of your software. Your software can be installed with a simple `snap install vscode --beta --classic` simplifying the installation instructions for your users.

Once the snap is released into the stable channel in the store, users no longer need to add the ```--beta``` and will appear in the desktop Software Center

### CODE REVIEW

File: docs/setup/linux.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ MetaDescription: Get Visual Studio Code up and running on Linux.
 
 ## Installation
 
+### Ubuntu and [snap supported distros](https://snapcraft.io/docs/core/install) 
+
+Install the snap on [snap supported](https://snapcraft.io/docs/core/install) systems from the store.
+
+    snap install vscode --beta --classic
```
Comment (Tyriar) [inline docs/setup/linux.md]: We would need to have the package named `code` as all our documentation references `code as the CLI, not `vscode`.

Comment (flexiondotorg) [inline docs/setup/linux.md]: The name `code` is already taken in the snap store. I'll see if I can claim it for Microsoft Visual Studio Code but for the time being I've updated the snap to support the alias `code`.

**Comment by Tyriar** [comment]
@flexiondotorg just following up on how the package would actually be distributed.

> I don't believe you explained how the packages are submitted? Not owning the upload ourselves would probably mean the release would lag behind the official release, no?

**Comment by Tyriar** [comment]
Never mind, just saw https://github.com/Microsoft/vscode/issues/5458#issuecomment-300742444 and replied there

**Comment by duttaditya18** [comment]
Getting the ownership of `code` in the snap store will take some time. In the mean time I think this can be merged.

**Comment by gregvanl** [comment]
snap package now officially supported. https://snapcraft.io/code

Documentation updated at https://code.visualstudio.com/docs/setup/linux#_snap

**Pull request closed without merging**